code/krepel/container/dictionary.d

```diff
@@ -66,6 +66,16 @@ struct Dictionary(K, V)
     }
   }
 
+  bool TryGet(InKeyType, OutValueType)(auto ref InKeyType Key, out OutValueType Value)
+  {
+    if(Contains(Key))
+    {
+      Value = this[Key];
+      return true;
+    }
+    return false;
+  }
+
```

Comment:
I think we should consolidate `TryGet` and `Contains` to a single implementation that returns a pointer to the element. This is the same as the `in`-operator in D, which is probably the *only* operator that you can't overload since it appears to be a convenient hack for associative arrays (AAs). Anyway, here's an AA example:

```d
int[string] StringToIntegerMap;
int* Answer = "answer" in StringToIntegerMap;
assert(Answer is null);
StringToIntegerMap["answer"] = 42;
assert(Answer !is null);
assert(*Answer == 42);
```

I imagine something like this for us:

```d
Dictionary!(string, int) StringToIntegerDict;
int* Answer = StringToIntegerDict.Get("answer");
// Answer = StringToIntegerDict["answer"]; // Would crash.
assert(Answer is null);
StringToIntegerDict["answer"] = 42;
Answer = StringToIntegerDict.Get("answer");
assert(Answer !is null);
assert(*Answer == 42);
```

`Contains`-like functionality is still there due to the implicit conversion of a pointer to `bool`:

```d
if(!StringToIntegerDict.Get("foo"))
{
  StringToIntegerDict["foo"] = /* Heavy operator to initialize a foo? */;
}
```